src/pkg/ingress/bindings/binding_config.go

```diff
@@ -32,23 +32,66 @@ func (d *DrainParamParser) FetchBindings() ([]syslog.Binding, error) {
 			continue
 		}
 
-		if d.defaultDrainMetadata && urlParsed.Query().Get("disable-metadata") == "true" {
-			b.OmitMetadata = true
-		} else if !d.defaultDrainMetadata && urlParsed.Query().Get("disable-metadata") == "false" {
-			b.OmitMetadata = false
-		} else {
-			b.OmitMetadata = !d.defaultDrainMetadata
-		}
-		if urlParsed.Query().Get("ssl-strict-internal") == "true" {
-			b.InternalTls = true
-		}
+		b.OmitMetadata = getOmitMetadata(urlParsed, d.defaultDrainMetadata)
+		b.InternalTls = getInternalTLS(urlParsed)
+		b.DrainData = getBindingType(urlParsed)
 
 		processed = append(processed, b)
 	}
 
 	return processed, nil
 }
 
+func getInternalTLS(url *url.URL) bool {
+	return url.Query().Get("ssl-strict-internal") == "true"
+}
+
+func getOmitMetadata(url *url.URL, defaultDrainMetadata bool) bool {
+	if defaultDrainMetadata && getRemoveMetadataQuery(url) == "true" {
+		return true
+	} else if !defaultDrainMetadata && getRemoveMetadataQuery(url) == "false" {
+		return false
+	} else {
+		return !defaultDrainMetadata
+	}
+}
+
+func getBindingType(u *url.URL) syslog.DrainData {
+	drainData := syslog.LOGS
+	switch u.Query().Get("drain-type") {
```

Comment:
@Benjamintf1 I know that with this handling of the `drain-type` and the addition of the `drain-data` parameter you've kept the things 100% backward compatible, but now we have two parameters which control the same thing: what will be forwarded by the syslog drain. That is confusing. Especially if a user specifies values for both parameters.

I was hoping that you will add `traces` as `drain-type` and `drain-type=all` will return logs + metrics + traces. IMHO this is an enhancement of the current `all` behavior and not a breaking change. If someone needs some drain type which is not covered by the built in combinations, they can create multiple drains. 

The other clean solution would be to remove the `drain-type` parameter and everything will be controlled the `drain-data` parameter. The users can define something like `drain-data=logs,metrics` or `drain-data=metrics,traces` and we parse the value and forward the envelope types which are specified with the parameter... It will a breaking change, but we could provide a script which will adjust the urls and replace the `drain-type` with an equivalent value for he `drain-data` parameter. If the `drain-type` parameter is missing we will add an `drain-data` value which is equivalent for the current `drain-type` behavior.

What do you think about this?

@ctlong Any thoughts on this?

Reply:
On a second thought, after sleeping over it, maybe we can stay backward compatible if we make the parameters mutually exclusive.

We could have something like:
```
if u.Query().Get("include-metrics-deprecated") != "" {
		return syslog.ALL
}

switch u.Query().Get("drain-data") {
      case "value" :
              return syslog_type
      ...
}

switch u.Query().Get("drain-type") {
      case "value" :
              return syslog_type
      ....
}
```

This way:
- if both parameters `drain-data` and `drain-type` are specified `drain-type` will have priority
- the currently active behavior can be still reached if only the `drain-type` parameter is specified, so no breaking changes.

I find the usage of both parameters still confusing in this case, but this might be a good compromise if we don't won't to introduce breaking changes. TBH I'm more in favor of breaking changes, and do the things clean with only one parameter.

Reply:
They're already mutually exclusive in the pr? The new drain-data overrides the old drain-type parameter. Also, given we havn't documented drain-type especially well, I believe most of the usage will be the old deprecated drain-cli usage, which shouldn't have mixed flags. 

include-metrics-deprecated also I guess overrides both? I didn't have a paticularly good reason to make it override everything, but I believe it should mostly be being used for log-cache, which we can replace with that all type flag at some point.

Reply:
@Benjamintf1 Thanks for the explanation. I've checked the code once again and saw that everything is as it should be.

Regarding the docs I will adjust the [Streaming app logs to log management services](https://docs.cloudfoundry.org/devguide/services/log-management.html) page and add the new parameter. 

Should I maybe write that the new `drain-data` parameter is the way to go and `drain-type` is used for backward compatibility when someone sets its value to `all`?